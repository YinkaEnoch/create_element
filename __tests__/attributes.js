const c = require("../src/index.ts");

test("create an header with id", () => {
	let el = c("h1", { id: "mast" });

	expect(el).toMatchSnapshot();
});

test("create a paragraph with class", () => {
	let el = c("p", { class: "text-center" });

	expect(el).toMatchSnapshot();
});

test("create an input element", () => {
	let el = c("input", {
		type: "text",
		id: "fullname",
		name: "fullname",
		required: true,
		placeholder: "Your full name",
	});

	expect(el).toMatchSnapshot();
});

test("create a label element", () => {
	let el = c("label", { for: "fullname" });
	expect(el).toMatchSnapshot();
});

test("create a div element with multiple classes", () => {
	let el = c("div", { class: "text-right text-primary bg-dark" });
	expect(el).toMatchSnapshot();
});

test("create a button with listener handler", () => {
	let el = c("button", {
		type: "button",
		onclick: () => console.log("Hello World!"),
		class: "btn-success",
	});
	expect(el).toMatchSnapshot();
});
