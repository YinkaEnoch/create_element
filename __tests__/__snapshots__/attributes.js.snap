// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a button with listener handler 1`] = `
<button
  class="btn-success"
  type="button"
/>
`;

exports[`create a div element with multiple classes 1`] = `
<div
  class="text-right text-primary bg-dark"
/>
`;

exports[`create a label element 1`] = `
<label
  for="fullname"
/>
`;

exports[`create a paragraph with class 1`] = `
<p
  class="text-center"
/>
`;

exports[`create an header with id 1`] = `
<h1
  id="mast"
/>
`;

exports[`create an input element 1`] = `
<input
  id="fullname"
  name="fullname"
  placeholder="Your full name"
  required="true"
  type="text"
/>
`;
